core/domain/suggestion_registry.py

```diff
@@ -67,12 +95,35 @@ class BaseSuggestion:
             reviewer.
     """
 
-    def __init__(self, status, final_reviewer_id):
+    # Here, we explicitly defined all the attributes that are used in
+    # BaseSuggestion because in `to_dict`, `get_score_type` and other
+    # methods too we are accessing these attributes but due to the lack
+    # of definition in main implementation the types of these attributes
+    # are not available which causes MyPy to throw undefined attribute
+    # error for all attributes that are used in BaseSuggestion. Thus to
+    # provide type-info to MyPy about these attributes, we defined them
+    # in `if` clause so that they don get executed at runtime.
```

Comment:
It can be executed at runtime but I'm not doing this before because I'm not sure about the test cases. But now all the cases are passing so I removed the `if MYPY` clause and updated the comment.